electrumsv/wallet_database/functions.py

```diff
@@ -2075,6 +2077,61 @@ def _write(db: Optional[sqlite3.Connection]=None) -> None:
     return db_context.post_to_thread(_write)
 
 
+async def read_reorged_transactions_async(db_context: DatabaseContext,
```

Comment:
All database reading should happen in the calling thread unless the reading operation is slow and will block the calling context. The database thread is for writing or combined writing and reading in the form of transactions that need to be atomic.